R/new.R

```diff
@@ -77,23 +84,27 @@ update_tibble_attrs <- function(x, ...) {
 
 guess_nrow <- function(x) {
   if (!is.null(.row_names_info(x, 0L))) {
-    .row_names_info(x, 2L)
+    list("nrow" = .row_names_info(x, 2L), "method" = "attribute")
   } else if (length(x) == 0) {
-    0L
+    list("nrow" = 0L, "method" = "detected empty list")
   } else {
-    NROW(x[[1L]])
+    list("nrow" = NROW(x[[1L]]), "method" = "first column length")
   }
 }
 
-validate_nrow <- function(x) {
+validate_nrow <- function(x, nrow_set_method) {
   # Validate column lengths, don't recycle
   lengths <- map_int(x, NROW)
-  first <- .row_names_info(x, 2L)
-
-  bad_len <- lengths != first
+  expected_nrow <- .row_names_info(x, 2L)
+  bad_len <- lengths != expected_nrow
   if (any(bad_len)) {
-    invalid_df_msg(
-      paste0("must be length ", first, ", not "), x, bad_len, lengths[bad_len]
+    vars <- names(x)[bad_len]
+    vars_len <- lengths[bad_len]
+    msg <- paste0("* Column `", vars, "` has length ", vars_len, "\n")
```

Comment:
What happens if we have too many inconsistent vars?

Reply:
Great catch -- I've fixed this so that if there are more than 5 inconsistent vars, the error msg is:

```
Error: Columns must have consistent lengths:
* Expected column length is 2 based on length of first column
* Column `b` has length 3
* Column `c` has length 3
* Column `d` has length 3
* Column `e` has length 3
* Column `f` has length 3
... and 5 more inconsistent columns 
```

Alternatively, the last line could be `... and the following inconsistent columns `g`, `h`, `i`, `j`, `k``. However, when there are 100s of inconsistent columns, this will still print long lines